js/PulsarFormComponent.js

```diff
@@ -86,31 +89,56 @@ PulsarFormComponent.prototype.initDatePickers = function () {
     const datepickers = this.$html.find('[data-datepicker="true"]');
     let defaultDateFormat = 'DD/MM/YYYY';
 
+    moment.locale(window.navigator.userLanguage || window.navigator.language);
+
     datepickers.each((index, element) => {
-        let dateFormat = element.getAttribute('data-format');
+        let formatKey = element.getAttribute('data-format');
+        let dateFormat = defaultDateFormat;
 
         // Check if data-format attribute exists and lowercase it
         // to eliminate different styles of writing issues
-        if (dateFormat !== null) {
-            dateFormat = dateFormat.toLowerCase();
+        if (formatKey !== null) {
+            formatKey = formatKey.toLowerCase();
         }
 
-        switch (dateFormat) {
+        switch (formatKey) {
             case 'us':
-                defaultDateFormat = 'MM/DD/YYYY';
+                dateFormat = 'MM/DD/YYYY';
                 break;
             case 'reverse':
-                defaultDateFormat = 'YYYY/MM/DD';
+                dateFormat = 'YYYY/MM/DD';
+                break;
+            case 'locale':
+                dateFormat = moment.localeData().longDateFormat('L');
                 break;
             default:
-                defaultDateFormat = 'DD/MM/YYYY';
+                dateFormat = 'DD/MM/YYYY';
```

Comment:
`dateFormat = defaultDateFormat` ?

Reply:
You need to create a local variable. Otherwise, you're altering the default. I think there might be a bug at the moment given multiple inputs:

```twig
{{ 
    form.date({
        'format': 'US',
    })
}}

{# This input would be in the US format, I think #}
{{ 
    form.date()
}}
```

Reply:
Perhaps not, because of the default. However, the local variable is needed for the event handler so it has a solid reference that won't change

Reply:
I'm not sure if I'm following, feels like what @Stanton is suggesting makes the most sense here? Keeping the `dateFormat` variable but then assigning it the value of the `defaultDateFormat` (line 90) in this default case

Reply:
Apologies! 🤦‍♂ I got confused. I agree @Stanton is correct now I've re-read this!